docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,469 @@
+
+A label set is a file that has actual labels each in a line. For example:
+
+```
+airplane
+automobile
+bird
+cat
+deer
+dog
+frog
+horse
+ship
+truck
+```
+
+Each dataset except for the label set must contain raw float array data and one-hot-encoded labels:
+
+```
+[float array for data1][label(on-hot encoded) for data1][float array][label]...
+```
+
+After preparing the label set and dataset, create `dataset` as follows:
+
+```c
+int status = ML_ERROR_NONE;
+ml_train_dataset_h dataset;
+
+char *res_path = app_get_resource_path();
+char train_path[1024];
+char label_prop[1024];
+
+snprintf(train_path, sizeof(train_path), "%s/training.dat", res_path);
+snprintf(label_prop, sizeof(label_prop), "label_data=%s/label.dat", res_path);
+free(res_path);
+
+status = ml_train_dataset_create_with_file(&dataset, train_path);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+
+status = ml_train_dataset_set_property(dataset, label_prop,
+                                   "buffer_size=9", NULL);
+status = ml_train_model_set_dataset(model, dataset);
+```
+
+`buffer_size` can be set for dataset.
+
+#### Set a dataset from a generator
+
+Creating a dataset from a generator function is also possible.
+
+1. Prepare a callback function:
+
+    ```c
+    /**
+    * @brief      get data which size is batch for train
+    * @param[out] data
+    * @param[out] label
+    * @param[out] last if the data is finished
+    * @param[in] user_data private data for the callback
+    * @retval status for handling error
+    */
+    int get_train_data(float **data, float **label, bool *last, void *user_data) {
+      /* code that fills data, label and last */
+      return ML_ERROR_NONE;
+    }
+    ```
+
+2. Create a dataset from the callback function:
+
+    ```c
+    int status = ML_ERROR_NONE;
+    ml_train_dataset_h dataset;
+
+    // validation and test callback can be omitted.
+    status = ml_train_dataset_create_with_generator(&dataset, get_train_data,
+                                                    NULL, NULL);
+    ```
+
+## Compile the model
+
+Compiling a model finalizes the model with loss.
+Once compiled, any modification to the properties of the model.
+Adding layers or changing the optimizer/dataset of the model will not be permitted as well:
+
+```c
+int status = ML_ERROR_NONE;
+
+status = ml_train_model_compile(model, "loss=cross", "batch_size=9", NULL);)
+```
+
+## Train the model
+
+Now, the model is ready to train. Train model as follows:
+
+```c
+int status = ML_ERROR_NONE;
+
+status = ml_train_model_run(model, "epochs=20", "save_path=model.bin", NULL);)
+```
+
+## Destroy the model
+
+After training, the model must be destroyed with `ml_train_model_destroy`.
+
+> [!NOTE]
+> `ml_train_model_add_layer`, `ml_train_set_optimizer`, `ml_train_set_dataset` transferrers ownership to `model`.
+> `layers`/`optimizers`/`dataset` that belongs to a `model` is also deleted.
```

Comment:
While reading the context, I had a doubt at this note. Should this context  be in note format? 
In my opinion, they can be in normal format like below.

After training, the model must be destroyed with `ml_train_model_destroy`.
`ml_train_model_add_layer`, `ml_train_set_optimizer`, and `ml_train_set_dataset` transfers ownership to `model`.
`layers`, `optimizers`, and `dataset` that belongs to a `model` is also deleted.

Reply:
This does not need to be in note format.  I have changed to note tag from the earlier suggestion in https://github.com/Samsung/tizen-docs/pull/1221#discussion_r480822286

@surbhig12, could you check this out?

Reply:
It is fine. Thank you

Reply:
Changed to the normal format.